lib/npm.js

```diff
@@ -15,6 +16,42 @@ var Npmrc = require('./npmrc');
 
 var defaultRegistry = 'https://registry.npmjs.org';
 
+// Test whether the contents of buffer is gzipped
+function isGzip(buffer) {
+  if (!buffer || buffer.length < 3) {
+    return false;
+  }
+  return buffer[0] === 0x1f && buffer[1] === 0x8b && buffer[2] === 0x08;
+}
+
+// Return a promise that gunzips the buffer *if* it is gzipped
+// Otherwise return a promise with the buffer as is. 
+function gunzip(buffer) {
+  if (isGzip(buffer)) {
+    return new Promise(function (resolve, reject) {
+      zlib.gunzip(buffer, function (err, decoded) {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(decoded);
+        }
+      });
+    });
+  } else {
+    return Promise.resolve(buffer);
+  }
+}
+
+// Create a readable stream from a buffer
+function bufferStream(buffer) {
+  return new stream.Readable({
+    read(size) {
+      this.push(buffer);
+      this.push(null);
+    }
```

Comment:
Any ideas re performance implications of either method?
On Mon, 08 Aug 2016 at 13:33, Morten Houston Ludvigsen <
notifications@github.com> wrote:

> In lib/npm.js <https://github.com/jspm/npm/pull/147#discussion_r73860465>:
>
> > +          resolve(decoded);
> > +        }
> > +      });
> > +    });
> > +  } else {
> > +    return Promise.resolve(buffer);
> > +  }
> > +}
> > +
> > +// Create a readable stream from a buffer
> > +function bufferStream(buffer) {
> > +  return new stream.Readable({
> > +    read(size) {
> > +      this.push(buffer);
> > +      this.push(null);
> > +    }
>
> If I use node-buffer-peek-stream
> <https://github.com/seangarner/node-buffer-peek-stream>, this code can
> disappear :-)
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/jspm/npm/pull/147/files/38cd6cebb1ba586f001ccdce47afebb3a1030cf0#r73860465>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAkiyoA15kp2lubQ6LWcOXVFZuKqtTJSks5qdxQCgaJpZM4Jeg1d>
> .
>